### PR TITLE
Stampit v4

### LINKED ISSUE #323: Reducing size even more
Recently @ericelliott posted a truly great concern that we are bloating stampit: https://github.com/stampit-org/stampit/pull/320#issuecomment-325812185

I agree, we should save those bytes (lines of code). I would even add that we should simplify stampit infrastructure because it was overly complexified with ES6->ES5 compilation.

Current stampit UMD build size is `5.26 KB`. By removing fancy ES6 stuff (plus some optimization techniques) I managed to reduce the size down to `4.0 KB`.

To be precise: `5393` -> `4128` bytes. It is `1,265` bytes or 23.5% less.

Simultaneously, this will fix #317, which was reported three times IIRC. (TL;DR: our own examples do not work in browsers.)

Also, this will reduce `devDepencencies` a bit.

Pinging all people from the #320 discussion: @sethlivingston @FredyC @Mustack @jyboudreau @ericelliott 

Should I proceed and submit the PR?

### PR DESCRIPTION
**Pull request by koresar**
Closes #323 #317 #304 #68 

Changes:
* Removed the previously deprecated `refs`.
* Removed `stampit/*` utility functions. Advice to use `@stamp/*` modules instead.
* `composers` are not marked as experimental any more.
* Stampit becomes compatible with all `@stamp/*` stamps.
* The `.min.js` bundle size reduced by ~50%, the gzipped size reduced by ~40%.
* Tests can now be manually run in a browser (by opening `./test/index.html`)
* Stampit source code is ES5 now instead of ES6.
* Fixed perf bottleneck in node.js 8 (new V8 engine).

In case of silence I'll merge this PR in a week.

### CODE REVIEW

**Comment by FredyC** [comment]
Phew, sorry @koresar, but I don't have a space to review this bulk of changes.

Honestly, I am not even sure why to bother with another stampit release when there is stamp monorepo. Why not to just have a `stampit@v4` to reference `@stamp/it` instead? Why two different code bases?

**Comment by koresar** [comment]
1. Stampit will become compatible with monorepo. Some were complaining
about it.
2. Size reduce. Good for bragging and websites perf.

The other changes have just happened along the way.

**Comment by FredyC** [comment]
Sorry, but that still does not explain why to keep two almost identical implementations. When I started the idea about monorepo I thought that eventually, stampit module will be just a redirect to this monorepo and after a while it might be even deprecated and removed.

Right now it's kinda unclear where is the difference. Is it possible to use extra modules from the monorepo with stampit? That's an example of a user question, I know they are both up to spec including composers, but in my opinion, it just makes things worse.

I can imagine it wasn't easy to accomplish size reduction you wanted, but it feels kinda OCD :) Do you know someone who actually complained that this lib is couple bytes larger than it could be? I am probably living in a different world where a size of the libs is so much higher and it does not really matter in overall.

Personally, I have kinda lost interest in stamps in overall. Perhaps I've never understood it that well to use it efficiently and without losing grasp over my codebase. In complex scenarios, it was just too messy for my taste. Not even mentioning adoption when working on the team and explaining everyone how it works.

If you feel like you want to go this direction then go ahead and merge and release it. I am not best to judge this anymore. Sorry friend.

**Comment by koresar** [comment]
Stampit have to jump to next major version only because it is not following the latest spec v1.5.
If we would vote to make "composers" part of the spec year ago stampit would have stayed v3.
Unfortunately, community took a careful decision to do baby steps - implement "composers" as an experimental feature first. 
So, stampit v4 if a necessary last baby step towards compatibility with the latest specification v1.5.

**Comment by AlexxNica** [comment]
@koresar Why did you remove the license?

File: license.txt [hunk 1]
```diff
@@ -1,21 +0,0 @@
-The MIT License (MIT)
```
Comment (AlexxNica) [inline license.txt]: Here

**Comment by AlexxNica** [comment]
@koresar Little tip: You could have rebased your commits before pushing and merging to avoid polluting the history.

Take a look at:
https://git-scm.com/docs/git-rebase
https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by koresar** [comment]
Totally forgot to squash before merging. My bad.

Licence was removed by accident. Will readd, or feel free to pr. Sorry
about that.

**Pull request merged**